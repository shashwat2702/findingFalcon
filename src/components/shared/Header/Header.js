import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import './Header.scss';
import hamburger from '../../../assets/hamburger.svg';

export class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activeLink: 'home',
    };
  }

  render() {
    return (
      <nav className="pageHeader">
        <div className="navLinks">
          <Link to="/" className="active">Home</Link>
          <Link to="/">News</Link>
          <Link to="/">Contact</Link>
          <Link to="/">About</Link>
          <img src={hamburger} alt="hamburger menu" />
        </div>
      </nav>
    );
  }
}
export default withRouter(Header);
