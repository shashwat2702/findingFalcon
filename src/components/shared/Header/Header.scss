.pageHeader {
    background-color: #1a1a1a;
    color: white;
    font-size: 2rem;
    display: flex;
    justify-content: flex-end;
    .active {
        background-color: #1ab188;
        color: white;
      }
    img {
        display: none;
    }
    .navLinks {
        padding-right: 4rem;
        display: flex;
        flex: 1;
        justify-content: flex-end;
    }
    .logoutButton {
        font-size: 1.5rem;
        height: 4rem;
        background-color: dodgerblue;
        border: 0;
        color: white;
    }
}
.pageHeader a {
    color: #f2f2f2;
    padding: 1rem;
    text-decoration: none;
    font-size: 1.5rem;
  }
.pageHeader a:hover {
background-color: #ddd;
color: black;
}

@media screen and (max-width: 600px) {
    .pageHeader a:not(:first-child) {display: none;}
    .pageHeader img {
        padding: 1rem;
        float: right;
        display: block;
    }
    .pageHeader{
        display: flex;
        justify-content: flex-start;
    }
    .pageHeader img {
        position: absolute;
        right: 0;
        top: 0;
    }
    .pageHeader a {
        float: none;
        display: block;
        text-align: left;
    }
    .active {
        width: 100%;
      }
}