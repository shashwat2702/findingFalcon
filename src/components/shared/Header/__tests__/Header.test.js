import React from 'react';
import renderer from 'react-test-renderer';
import { Header } from '../Header';

jest.mock('react-router-dom');

describe('HomePage snapshot', () => {
  it('should render correctly', () => {
    const HeaderSnap = renderer.create(<Header />);
    const HeaderJson = HeaderSnap.toJSON();
    expect(HeaderJson).toMatchSnapshot();
  });
});
