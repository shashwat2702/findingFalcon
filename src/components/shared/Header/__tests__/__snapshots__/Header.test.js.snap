// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage snapshot should render correctly 1`] = `
<nav
  className="pageHeader"
>
  <div
    className="navLinks"
  >
    <img
      alt="hamburger menu"
      src="hamburger.svg"
    />
  </div>
</nav>
`;
